python/ray/_private/gcs_utils.py

```diff
@@ -302,6 +319,114 @@ def pin_runtime_env_uri(self, uri: str, expiration_s: int) -> None:
             )
 
 
+class GcsAioClient:
+    def __init__(
+        self,
+        channel: Optional[GcsChannel] = None,
+        address: Optional[str] = None,
+    ):
+        if channel is None:
+            assert isinstance(address, str)
+            channel = GcsChannel(gcs_address=address, aio=True)
+        assert isinstance(channel, GcsChannel)
+        assert channel._aio is True
+        self._channel = channel
+        self._connect()
+
+    def _connect(self):
+        self._channel.connect()
+        self._kv_stub = gcs_service_pb2_grpc.InternalKVGcsServiceStub(
+            self._channel.channel()
+        )
+
+    async def internal_kv_get(
+        self, key: bytes, namespace: Optional[bytes], timeout: Optional[float] = None
+    ) -> Awaitable[bytes]:
+        logger.debug(f"internal_kv_get {key} {namespace}")
+        req = gcs_service_pb2.InternalKVGetRequest(namespace=namespace, key=key)
+        reply = await self._kv_stub.InternalKVGet(req, timeout=timeout)
+        if reply.status.code == GcsCode.OK:
+            return reply.value
+        elif reply.status.code == GcsCode.NotFound:
+            return None
+        else:
+            raise RuntimeError(
+                f"Failed to get value for key {key} "
+                f"due to error {reply.status.message}"
+            )
+
+    async def internal_kv_put(
+        self,
+        key: bytes,
+        value: bytes,
+        overwrite: bool,
+        namespace: Optional[bytes],
+        timeout: Optional[float] = None,
+    ) -> Awaitable[int]:
+        logger.debug(f"internal_kv_put {key} {value} {overwrite} {namespace}")
+        req = gcs_service_pb2.InternalKVPutRequest(
+            namespace=namespace,
+            key=key,
+            value=value,
+            overwrite=overwrite,
+        )
+        reply = await self._kv_stub.InternalKVPut(req, timeout=timeout)
+        if reply.status.code == GcsCode.OK:
+            return reply.added_num
+        else:
```

Comment:
What about the NotFound case?

Reply:
I think the module lacks good documentation here. Let me put it in another PR.

NotFound won't returned here since this is PUT operations and the return is about how many keys were added. Here the return is wired, but this is a legacy reason since the first version of internal kv defines the return in this way and we need to keep backward compatible for now.